### DISCUSSION TITLE
CGNS_DOTVERS - incorrect version it seems

### DISCUSSION THREAD

**Discussion by mennodeij** [CGNS/CGNS#357] [CGNStalk]
When trying to deduce the version of the library, the `FindCGNS.cmake` file from e.g. VTK, ParaView use the `CGNS_DOTVERS` version. It seems however, that this version is incorrect, when I download CGNS 4.1.2, the `CGNS_DOTVERS` is 4.10; for CGNS library version 4.2.0 `CGNS_DOTVERS` is 4.20 ; is there some logic here? Is there another way to find out the true triple-number library version?

**Comment by MicK7** [comment]
CGNS_DOTVERS should be the correct version of the library.
For CGNS 4.1.2, the expected value is 4.12 while for 4.2.0 it is 4.20. May be it was not correctly updated before packaging the release.
CGNS_DOTVERS is also used by CGNS cmake main script to determine the cpack version.

**Comment by mennodeij** [comment]
Ok, clear, the `CGNS_DOTVERS` constant should have been 4.12 in version 4.1.2.
Currently, cgnslib.h for that version reads
```
#define CGNS_VERSION 4100
#define CGNS_DOTVERS 4.10
```

Futhermore, the VTK/ParaView version requirements must be changed if the versioning is like this. I have started an issue for this at https://gitlab.kitware.com/vtk/vtk/-/issues/18654
